lib/recurly/pricing/index.js

```diff
@@ -243,14 +252,18 @@ Pricing.prototype.coupon = function (couponCode, done) {
 
   return new PricingPromise(function (resolve, reject) {
     if (!self.items.plan) return reject(errors('missing-plan'));
-    if (coupon) {
-      if (coupon.code === couponCode) return resolve(coupon);
-      else self.remove({ coupon: coupon.code });
+    if (coupon) self.remove({ coupon: coupon.code });
+
+    // TODO: Should this emit an 'unset' event?
+    // Need to consider other objects as well
+    if (!couponCode) {
+      debug('unset.coupon');
+      delete self.items.coupon;
+      return resolve();
     }
-    if (!couponCode) return resolve();
 
     self.recurly.coupon({ plan: self.items.plan.code, coupon: couponCode }, function (err, coupon) {
-      if (err && err.code !== 'not_found') return reject(err);
+      if (err && err.code !== 'not-found') return reject(err);
```

Comment:
I'd like to follow up on this PR with some reconsideration of how we handle not-found coupons. Right now we ignore the error completely, but I think we should probably emit an event that merchants can listen to. See #176